### PR TITLE
bpf: make setting fifo policy non-fatal when probing hz

### PR DESCRIPTION
**Pull request by borkmann**
See commit msg.

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

File: bpf/cilium-probe-kernel-hz.c [hunk 1]
```diff
@@ -33,13 +33,26 @@ static int pin_to_cpu(int cpu)
...
+static void fix_priority(void)
+{
+	int ret_fifo = pin_scheduler(SCHED_FIFO);
+
+	if (ret_fifo) {
+		int ret_nice = setpriority(PRIO_PROCESS, 0, -20);
+
+		if (ret_fifo || ret_nice)
+			printf("/* Note, could not set %s for probing KERNEL_HZ. */\n",
+			       ret_fifo && ret_nice ? "fifo and nice" : "fifo");
```
Comment (pchaigno) [inline bpf/cilium-probe-kernel-hz.c]: I tested this again with artificial load (processes with high priority on same CPU). When we can't set both fifo and nice, the program fails a lot more often, most of the time with a `probed jiffies mismatch` but also sometimes just with an incorrect HZ value (e.g., 300 or 1000).

I don't know what impact an incorrect HZ value would have on the datapath, but maybe we could also add a threshold on the warp value (when the HZ value is incorrect, warp is always 2-3-digits long)? Then again, I had to have several processes with highest priority running on the same core to see that happening, so probably quite unlikely.

**Review by pchaigno — APPROVED** [review]
@borkmann Could you also add the executable to `bpf/.gitignore`?

File: bpf/cilium-probe-kernel-hz.c (hunk 1 above)
Comment (borkmann) [reply bpf/cilium-probe-kernel-hz.c]: Ok, to be on safe side, I've reworked everything. Will post in a bit.

**Comment by borkmann** [comment]
test-me-please

**Review by pchaigno — APPROVED** [review]
LGTM!

**Comment by borkmann** [comment]
test-with-kernel

**Comment by nebril** [comment]
retest-4.19

**Pull request merged**